mcs/class/System/Test/System.Diagnostics/PerformanceCounterTest.cs

```diff
@@ -0,0 +1,331 @@
+		internal PerformanceCounterCategory pcc;
+
+		internal Common ()
+		{
+			guid = Guid.NewGuid ().ToString ("N");
+			name = guid + "_Counter";
+			category = name + "_Category";
+			counterDataCollection = new CounterCreationDataCollection ();
+
+			// Add the counter.
+			counter = new CounterCreationData ();
+			counter.CounterType = PerformanceCounterType.AverageCount64;
+			counter.CounterName = name;
+			counterDataCollection.Add (counter);
+
+			// Add the base counter.
+			counterBase = new CounterCreationData ();
+			counterBase.CounterType = PerformanceCounterType.AverageBase;
+			baseName = name + "Base";
+			counterBase.CounterName = baseName;
+			counterDataCollection.Add (counterBase);
+
+			// Create the category.
+			PerformanceCounterCategory.Create (category, "description",
+				PerformanceCounterCategoryType.SingleInstance, counterDataCollection);
+
+			counterSample = new PerformanceCounter (category, name, false);
+
+			counterSample.RawValue = 0;
+
+			pcc = new PerformanceCounterCategory (category);
+		}
+	}
+
+	[TestFixture]
+	public class PerformanceCounterTest
+	{
+		Common a = new Common ();
+
+		void WriteLine (string arg)
+		{
+			if (!verbose)
+				return;
+			Console.WriteLine (arg);
+		}
+
+		[Test]
+		public void PerformanceCounterCategory_CreateCategory ()
+		{
+			WriteLine ("PerformanceCounterCategory_CreateCategory");
+
+			var guid = Guid.NewGuid ().ToString ("N");
+			string name = "AverageCounter64Sample" + guid;
+
+			Assert.False (PerformanceCounterCategory.Exists (name + "Category"));
+
+			CounterCreationDataCollection counterDataCollection = new CounterCreationDataCollection ();
+
+			// Add the counter.
+			CounterCreationData averageCount64 = new CounterCreationData ();
+			averageCount64.CounterType = PerformanceCounterType.AverageCount64;
+			averageCount64.CounterName = name;
+			counterDataCollection.Add (averageCount64);
+
+			// Add the base counter.
+			CounterCreationData averageCount64Base = new CounterCreationData ();
+			averageCount64Base.CounterType = PerformanceCounterType.AverageBase;
+			averageCount64Base.CounterName = name + "Base";
+			counterDataCollection.Add (averageCount64Base);
+
+			// Create the category.
+			PerformanceCounterCategory.Create (name + "Category",
+			"Demonstrates usage of the AverageCounter64 performance counter type.",
+			PerformanceCounterCategoryType.SingleInstance, counterDataCollection);
+
+			Assert.True (PerformanceCounterCategory.Exists (name + "Category"));
+
+			WriteLine ("PerformanceCounterCategory_CreateCategory end");
+		}
+
+		[Test]
+		public void PerformanceCounter_CreateCounter_Count0 ()
+		{
+			WriteLine ("PerformanceCounter_CreateCounter_Count0 start");
+			Assert.AreEqual (0, a.counterSample.RawValue);
+			a.counterSample.Increment ();
+			Assert.AreEqual (1, a.counterSample.RawValue);
+			WriteLine ("PerformanceCounter_CreateCounter_Count0 end");
+		}
+
+		bool verbose = Environment.GetEnvironmentVariable ("V") != null;
```

Comment:
Why is this in the middle of the class? Why can't it more simply be at the beginning where all the fields are declared?